tools/wiki_sync.py

```diff
@@ -0,0 +1,173 @@
+
+    return sidebar
+
+
+def write_sidebar_file(sidebar: Dict[str, List[DocumentationFile]]) -> None:
+    sidebar_format = "'''Contents'''\n"
+    sidebar_format += "*'''[[ Home | Home ]]'''\n"
+
+    depth = 1  # helps track the level of nesting in the wiki sidebar
+    for category, doc_files in sidebar.items():
+        if category == "home":
+            continue
+
+        subcategories = category.split("/")
+        for subcategory in subcategories:
+            sidebar_format += ("*" * depth) + f"'''{subcategory.title()}'''\n"
+            depth += 1
+
+        for doc in doc_files:
+            sidebar_format += (
+                "*" * depth
+            ) + f"[[ {doc.file_name} | {doc.display_name} ]]\n"
+
+        # reset depth for the next category
+        depth = 1
+
+    with open(f"{WIKI_DIR}/_Sidebar.mediawiki", "w") as f:
+        f.write(sidebar_format)
+
+
+def write_wiki_pages(sidebar: Dict[str, List[DocumentationFile]]) -> None:
+    # open md files in the repo an prepare for migration process
+    
+    for category, doc_files in sidebar.items():
+        for doc_file in doc_files:
+            with open(doc_file.path) as source:
+                target = f"{WIKI_DIR}/{doc_file.file_name}.md"
+                if category == "home":
+                    target = f"{WIKI_DIR}/Home.md"
+
+                with open(target, "w") as dest:
+                    dest.write(source.read())
+
+
+def run_wiki_migration():
+    os.makedirs(TEMP_WIKI, exist_ok=True)
+
+    # repo env variables
+    secret = os.environ["USER_TOKEN"]
+    user_name = os.environ["USER_NAME"]
+    user_email = os.environ["USER_EMAIL"]
+    owner = os.environ["OWNER"]
+    repo_name = os.environ["REPOSITORY_NAME"]
+
+    # init, pull, delete
+    subprocess.call(["git", "init"], cwd=TEMP_WIKI)
+    subprocess.call(["git", "config", "user.name", f"{user_name}"], cwd=TEMP_WIKI)
+    subprocess.call(["git", "config", "user.email", f"{user_email}"], cwd=TEMP_WIKI)
+    subprocess.call(
+        ["git", "pull", f"https://{secret}@github.com/{owner}/{repo_name}.wiki.git"],
+        cwd=TEMP_WIKI,
+    )
+
+    # sync the new and old wiki files
+    subprocess.call(
+        [
+            "rsync",
+            "-av",
+            "--delete",
+            "tools/wiki/",
+            "tools/temp_wiki",
+            "--exclude",
+            ".git",
+        ]
+    )
+
+    # add, config, commit and push
+    subprocess.call(["git", "add", "."], cwd=TEMP_WIKI)
+    sha_hash = (
+        subprocess.check_output(
+            ["git", "rev-list", "--max-count=1", "HEAD"], cwd=TEMP_WIKI
+        )
+        .decode()
+        .strip()
+    )
+    subprocess.call(
+        ["git", "commit", "-m", f"update wiki content - sha: {sha_hash}"], cwd=TEMP_WIKI
+    )
+    subprocess.call(
+        [
+            "git",
+            "remote",
+            "add",
+            "master",
+            f"https://{secret}@github.com/{owner}/{repo_name}.wiki.git",
```

Comment:
Similarly, this change is necessary:

```suggestion
            f"https://{os.environ["GITHUB_ACTOR"]}:{os.environ["GITHUB_TOKEN"]}@github.com/{owner}/{repo_name}.wiki.git",
```